staging/src/k8s.io/apimachinery/third_party/forked/golang/json/fields.go

```diff
@@ -52,9 +52,41 @@ func LookupPatchMetadata(t reflect.Type, jsonField string) (reflect.Type, string
 		}
 		patchStrategy := tjf.Tag.Get("patchStrategy")
 		patchMergeKey := tjf.Tag.Get("patchMergeKey")
-		return tjf.Type, patchStrategy, patchMergeKey, nil
+		hasReplacekeys, otherPatchStrategy, err := extractReplaceKeysPatchStrategy(patchStrategy)
+		if err != nil {
+			return nil, false, "", "", err
+		}
+		return tjf.Type, hasReplacekeys, otherPatchStrategy, patchMergeKey, nil
+	}
+	return nil, false, "", "", fmt.Errorf("unable to find api field in struct %s for the json field %q", t.Name(), jsonField)
+}
+
+// extractReplaceKeysPatchStrategy process patch strategy, which is a string may contains multiple
+// patch strategies seperated by "|". It returns a boolean var indicating if it has replaceKeys strategies
+// and a string or other strategy.
+func extractReplaceKeysPatchStrategy(patchStrategy string) (bool, string, error) {
+	replaceKeysPatchStrategy := "replaceKeys"
+	strategies := strings.Split(patchStrategy, ",")
+	if len(strategies) == 1 {
+		singleStrategy := strategies[0]
+		switch singleStrategy {
+		case replaceKeysPatchStrategy:
+			return true, "", nil
+		default:
+			return false, singleStrategy, nil
+		}
+	}
+	if len(strategies) > 2 {
+		return false, "", fmt.Errorf("Unexpected patch strategy: %q\n", patchStrategy)
+	}
+	switch {
+	case strategies[0] == replaceKeysPatchStrategy:
+		return true, strategies[1], nil
+	case strategies[1] == replaceKeysPatchStrategy:
+		return true, strategies[0], nil
+	default:
+		return false, "", fmt.Errorf("Unexpected patch strategy: %q\n", patchStrategy)
```

Comment:
> So we allow at most 2 patch strategies, and if there are 2 strategies, one of them needs to be "replaceKeys". Correct?

In current scope, yes. You can find "replaceKeys,merge" in the unit test.
WDYT if I make it more general by returning a string list for strategy?
```
func LookupPatchMetadata(t reflect.Type, jsonField string) (reflect.Type, []string, string, error)
```